spec_opts_unix.go

```diff
@@ -224,7 +224,7 @@ func WithUserNamespace(container, host, size uint32) SpecOpts {
 
 // WithRemappedSnapshot creates a new snapshot and remaps the uid/gid for the
 // filesystem to be used by a container with user namespaces
-func WithRemappedSnapshot(id string, i Image, uid, gid uint32) NewContainerOpts {
+func WithRemappedSnapshot(id string, i Image, uid, gid uint32, readonly bool) NewContainerOpts {
```

Comment:
Rather than take a flag, could we have `WithRemappedSnapshot` and `WithRemappedSnapshotView`. The "view" is a separate type, not a flag change, since it may be shared among others using the same snapshot.

The `readonly` bool flags are dicey to get right and easy to switch up (ie `writeable` as a variable name).